scripts/apps/monitoring/services/CardsService.ts

```diff
@@ -76,33 +94,42 @@ export function CardsService(search, session, desks, $location) {
 
     function filterQueryByCardType(query, queryParam, card: ICard) {
         let deskId;
+        const extensionSection = getExtensionSections();
+
+        extensionSection.then((response) => {
+            const data = response.find((res: IPersonalSpaceSection) => res?.id === card.type);
```

Comment:
> `res?.id`

Is it possible that `res.id` would be `null`?

I thought it wasn't, because compiler forces the extension to set a string and inside `getExtensionSections` you have a default value of an empty array which gets filtered out by `flattenDeep`.

Reply:
Previously I was getting it, but after making the latest changes (API return array), I think I can use flatMap and that will remove the empty array.

Reply:
```
_.flattenDeep([1, [2, 3], []])
> (3) [1, 2, 3]
```

`flattenDeep` removes empty arrays too. You wasn't using a default value before and that's why you were getting it.

Reply:
got it.